src/palace/manager/sqlalchemy/util.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Generator
+from typing import Literal, TypeVar
+
+from contextlib2 import contextmanager
+from psycopg2._range import NumericRange
+from sqlalchemy import text
+from sqlalchemy.engine import Connection
+from sqlalchemy.exc import IntegrityError, MultipleResultsFound, NoResultFound
+from sqlalchemy.orm import Session
+
+# This is the lock ID used to ensure that only one circulation manager
+# initializes or migrates the database at a time.
+LOCK_ID_DB_INIT = 1000000001
+
+
+@contextmanager
```

Comment:
broken out of `model/__init__.py`. This is where the bulk of the non-import content in that file was moved to.